skyplane/planner/planner.py

```diff
@@ -105,12 +170,18 @@ def plan(self, jobs: List[TransferJob]) -> TopologyPlan:
             assert [iface.region_tag() for iface in job.dst_ifaces] == dst_region_tags, "Add jobs must have same destination set"
 
         plan = TopologyPlan(src_region_tag=src_region_tag, dest_region_tags=dst_region_tags)
-        # TODO: use VM limits to determine how many instances to create in each region
+
+        # Dynammically calculate n_instances based on quota limits - do_parallel returns
+        # tuples of (vcpus, n_instances)
+        vm_info = do_parallel(self._calculate_vm_types, [src_region_tag] + dst_region_tags)  # type: ignore
+        vm_types = {v[0]: Planner._vcpus_to_vm(cloud_provider=v[0].split(":")[0], vcpus=v[1][0]) for v in vm_info}  # type: ignore
+        n_instances = min(v[1][1] for v in vm_info)  # type: ignore
```

Comment:
I am taking the minimum so that the source and the destination have the same number of instances. Example:
```
vm_classes = (96, 32, 8)
gcp_quota = 24
aws_quota = 5
```
gcp would select n2-standard-8 and would want to launch 3 of these (3 * 8 = 24) while was would select m5.xlarge and would be able to launch only one instance. In this situation, n_instances = min(1, 3) = 1